hera_pspec/grouping.py

```diff
@@ -494,20 +769,91 @@ def fold_spectra(uvp):
             right = uvp.data_array[spw][:, Ndlys//2+1:, :]
             uvp.data_array[spw][:, Ndlys//2+1:, :] = np.mean([left, right], axis=0)
             uvp.data_array[spw][:, :Ndlys//2, :] = 0.0
+            left = uvp.data_array_q[spw][:, 1:Ndlys//2, :][:, ::-1, :]
+            right = uvp.data_array_q[spw][:, Ndlys//2+1:, :]
+            uvp.data_array_q[spw][:, Ndlys//2+1:, :] = np.mean([left, right], axis=0)
+            uvp.data_array_q[spw][:, :Ndlys//2, :] = 0.0
             uvp.nsample_array[spw] *= 2.0
 
             # fold covariance array if it exists.
-            if hasattr(uvp,'cov_array'):
-                leftleft = uvp.cov_array[spw][:, 1:Ndlys//2, 1:Ndlys//2, :][:, ::-1, ::-1, :]
-                leftright = uvp.cov_array[spw][:, 1:Ndlys//2, Ndlys//2+1:, :][:, ::-1, :, :]
-                rightleft = uvp.cov_array[spw][:, Ndlys//2+1: , 1:Ndlys//2, :][:, :, ::-1, :]
-                rightright = uvp.cov_array[spw][:, Ndlys//2+1:, Ndlys//2+1:, :]
-                uvp.cov_array[spw][:, Ndlys//2+1:, Ndlys//2+1:, :] = .25*(leftleft\
-                                                                         +leftright\
-                                                                         +rightleft\
-                                                                         +rightright)
-                uvp.cov_array[spw][:, :Ndlys/2, :, :] = 0.0
-                uvp.cov_array[spw][:, :, :Ndlys/2, : :] = 0.0
+            #if hasattr(uvp,'cov_array'):
+            #    leftleft = uvp.cov_array[spw][:, 1:Ndlys//2, 1:Ndlys//2, :][:, ::-1, ::-1, :]
+            #    leftright = uvp.cov_array[spw][:, 1:Ndlys//2, Ndlys//2+1:, :][:, ::-1, :, :]
+            #    rightleft = uvp.cov_array[spw][:, Ndlys//2+1: , 1:Ndlys//2, :][:, :, ::-1, :]
+            #    rightright = uvp.cov_array[spw][:, Ndlys//2+1:, Ndlys//2+1:, :]
+            #    uvp.cov_array[spw][:, Ndlys//2+1:, Ndlys//2+1:, :] = .25*(leftleft\
+            #                                                             +leftright\
+            #                                                             +rightleft\
+            #                                                             +rightright)
+            #    uvp.cov_array[spw][:, :Ndlys/2, :, :] = 0.0
+            #    uvp.cov_array[spw][:, :, :Ndlys/2, : :] = 0.0
+            if hasattr(uvp,'cov_array_real'):
```

Comment:
another example where an in-line function would help.

Reply:
Thanks!